shared/indexer/indexer_test.go

```diff
@@ -0,0 +1,269 @@
+package indexer
+
+import (
+	"bytes"
+	"log"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+func FuzzTxIndexer(f *testing.F) {
+	operations := []string{
+		"GetByHash",
+		"GetByHeight",
+		"GetBySender",
+		"GetByRecipient",
+	}
+	numOperationTypes := len(operations)
+	numberOfOperations := 100
+	for i := 0; i < numberOfOperations; i++ {
+		f.Add(operations[rand.Intn(numOperationTypes)])
+	}
+	indexer, err := NewMemTxIndexer()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer indexer.Close()
+
+	f.Fuzz(func(t *testing.T, op string) {
+		// seed random
+		rand.Seed(int64(time.Now().Nanosecond()))
+		// set height ordering to descending 50% of time
+		var descending bool
+		if rand.Intn(2) == 0 {
+			descending = true
+		}
+		// select a height 0 - 9 to index
+		height := int64(rand.Intn(10))
+		// get index
+		heightResult, err := indexer.GetByHeight(height, descending)
+		require.NoError(t, err)
+		heightIndex := len(heightResult)
```

Comment:
Why is the index the number of results? Ditto bellow.

Is it meant to represent its position?

I saw the switch statement below but still find it hard to follow.